### PR TITLE
 TIP-863: Move Product to Enrichment sub context

### PR DESCRIPTION
**Pull request by jjanvier**
This moves the model, the abstract class and the interface. Other small PRs to come for all the "enrichment models".

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Y
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: STANDARD_FORMAT.md [hunk 1]
```diff
@@ -495,7 +495,7 @@ The structure of the array is composed as below:
 
 Each element in the array "groups" represents the *code* of the *Pim\Component\Catalog\Model\GroupInterface*
 
-Each element in the array "products" represents the *identifier* of the *Pim\Component\Catalog\Model\ProductInterface*
```
Comment (pierallard) [inline STANDARD_FORMAT.md]: Not related to this PR particulary, but all the doc has to be updated with these new namespaces at a moment...

File: src/Pim/Bundle/AnalyticsBundle/Doctrine/Query/CountVariantProduct.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 use Akeneo\Tool\Component\StorageUtils\Repository\CountableRepositoryInterface;
 use Doctrine\ORM\EntityManagerInterface;
-use Pim\Component\Catalog\Model\ProductInterface;
+use Akeneo\Pim\Enrichment\Component\Product\Model\ProductInterface;
```
Comment (pierallard) [inline src/Pim/Bundle/AnalyticsBundle/Doctrine/Query/CountVariantProduct.php]: php cs will fail :)

File: src/Pim/Component/Catalog/Normalizer/Indexing/ProductAndProductModel/ProductPropertiesNormalizer.php [hunk 3]
```diff
@@ -163,7 +163,7 @@ private function getAllParentsValues($productModel, array $context) : array
     }
 
     /**
-     * @param ProductInterface $product
+     * @param \Akeneo\Pim\Enrichment\Component\Product\Model\ProductInterface $product
```
Comment (arnolanglade) [inline src/Pim/Component/Catalog/Normalizer/Indexing/ProductAndProductModel/ProductPropertiesNormalizer.php]: you should revert that

File: src/Pim/Bundle/AnalyticsBundle/Doctrine/Query/CountVariantProduct.php (hunk 2 above)
Comment (jjanvier) [reply src/Pim/Bundle/AnalyticsBundle/Doctrine/Query/CountVariantProduct.php]: yep!
we'll merge a lot of little PR in a branch
I'll do it only once in the "feature" branch

File: src/Pim/Component/Catalog/Normalizer/Indexing/ProductAndProductModel/ProductPropertiesNormalizer.php (hunk 3 above)
Comment (jjanvier) [reply src/Pim/Component/Catalog/Normalizer/Indexing/ProductAndProductModel/ProductPropertiesNormalizer.php]: why?

File: STANDARD_FORMAT.md (hunk 1 above)
Comment (jjanvier) [reply STANDARD_FORMAT.md]: yes, you're absolutely right

Comment (jjanvier) [reply STANDARD_FORMAT.md]: (we have a card for that)
(we will apply our sed to the doc)

**Pull request merged**